CIS_controls/create_cis_catalogs_profiles.py

```diff
@@ -0,0 +1,174 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script to convert CIS controls spreadsheet into catalogs and profiles based on implementation group."""
+
+import argparse
+import datetime
+import logging
+import pathlib
+import sys
+from typing import List, Optional
+from uuid import uuid4
+
+from ilcli import Command
+
+import pandas as pd
+
+import trestle.oscal.catalog as oscat
+import trestle.oscal.common as oscommon
+import trestle.oscal.profile as ospro
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
+
+def run(input_xls: pathlib.Path, output_directory: pathlib.Path, cis_version: str) -> int:
+    """Run the conversion process."""
+    # To be reviewed - this awkward double read is annoying.
+    excel_handler = pd.ExcelFile(input_xls)
+    df = None
+    for key in excel_handler.sheet_names:
+        if 'controls' in str(key).lower():
+            sheet_name = key
+
+    df = pd.read_excel(input_xls, sheet_name=sheet_name, header=0, dtype=str)
+    # goal: output single catalog of controls and subcontrols
+    # also - output separate profiles for each of the implementation groups
+
+    # for profiles / implementation groups
+    grp1: List[str] = []
+    grp2: List[str] = []
+    grp3: List[str] = []
```

Comment:
Could be cleaner as array of groups as you probably realize.  This may be from my original script?  No biggy